pkg/controllers/disruption/drift_test.go

```diff
@@ -352,91 +339,59 @@ var _ = Describe("Drift", func() {
 						},
 					},
 				})
+				for i := range ncs {
+					ncs[i].StatusConditions().SetTrue(v1.ConditionTypeDrifted)
+					ExpectApplied(ctx, env.Client, ncs[i], ns[i])
+				}
 				nodeClaims = append(nodeClaims, ncs...)
 				nodes = append(nodes, ns...)
-			}
-			ExpectApplied(ctx, env.Client, nodePool)
-			for i := 0; i < len(nodeClaims); i++ {
-				nodeClaims[i].StatusConditions().SetTrue(v1.ConditionTypeDrifted)
-				ExpectApplied(ctx, env.Client, nodeClaims[i], nodes[i])
-			}
-
-			// inform cluster state about nodes and nodeclaims
-			ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, nodes, nodeClaims)
-			ExpectSingletonReconciled(ctx, disruptionController)
-
-			for _, np := range nps {
-				metric, found := FindMetricWithLabelValues("karpenter_nodepools_allowed_disruptions", map[string]string{
-					"nodepool": np.Name,
-				})
-				Expect(found).To(BeTrue())
-				Expect(metric.GetGauge().GetValue()).To(BeNumerically("==", 2))
-			}
-
-			// Execute the command in the queue, only deleting 20 nodes
-			ExpectSingletonReconciled(ctx, queue)
-			Expect(len(ExpectNodeClaims(ctx, env.Client))).To(Equal(10))
-		})
-		It("should allow all nodes from each nodePool to be deleted", func() {
```

Comment:
This would now happen in emptiness because the test used to have empty && drifted nodes. There is a test that covers disruptions across multiple nodepools.